src/com/amazon/ion/benchmark/ParseAndCompareBenchmarkResults.java

```diff
@@ -57,110 +38,110 @@ public class ParseAndCompareBenchmarkResults {
+        double pValue = tTest.tTest(before, after);
+        // Calculate means of both datasets
+        double meanBefore = StatUtils.mean(before);
+        double meanAfter = StatUtils.mean(after);
+        // Calculate the difference in means (regression value)
+        double regressionValue = (meanAfter - meanBefore) / meanBefore;
+        // Two-sample T-test is a hypothesis test used to compare the means of two independent groups.
+        // Null hypothesis states that there is no difference between two groups.
+        // P-value represents the probability of null hypothesis is true.
+        // Threshold value 0.05 is a common convention. We might adjust the value if we'd like to reduce the false positives/negatives.
+        // If the calculated p-value is greater than 0.05, then the result failed to reject the null hypothesis which means there isn't enough statistical evidence to say two groups are different.
+        // If the p-value is smaller than 0.05, then the result is considered statistically significant and the null hypothesis would be rejected. There is strong evidence that two groups of sample data are different.
+        if (pValue < 0.05) {
+            return regressionValue;
+        } else {
+            return 0;
         }
-        BigDecimal threshold = calculateDifference(Collections.max(rawData), Collections.min(rawData));
-        return threshold;
     }
 
     /**
-     * Get score of specific aspect from set (speed | heap usage | serialized size | gc.allocated.rate) after parsing the benchmark result.
-     *
-     * @param benchmarkResultFilePath is the path of benchmark result file.
-     * @param keyWord                 from set (speed | Heap usage | Serialized size | ·gc.alloc.rate) specifies which score will be extracted from the benchmark result.
-     * @return the score of specific aspect in BigDecimal format.
-     * @throws Exception if error occurs when reading Ion Data.
+     * This method is used for preprocessing the raw data from benchmark results to make sure the data is ready to be fed into the regression-detection process.
+     * @param benchmarkResult represents the file path of the benchmark result.
+     * @param keyWord represents the metric that will be extracted from the benchmark file.
+     * @return the processed data which is ready to be fed into two-sample T-test algorithm.
+     * @throws Exception if there's error occurred reading the benchmark result.
      */
-    public static BigDecimal getScore(String benchmarkResultFilePath, String keyWord) throws Exception {
-        IonStruct scoreStruct = parseScore(benchmarkResultFilePath, keyWord);
-        IonValue score = scoreStruct.get(SCORE);
-        if (score.getType().equals(IonType.FLOAT)) {
-            IonFloat scoreFloat = (IonFloat) score;
-            return scoreFloat.bigDecimalValue();
+    public static double[] preProcess(String benchmarkResult, String keyWord) throws Exception {
```

Comment:
We should use a more descriptive name than `preProcess`. How about `loadKeywordSpecificBenchmarkResults`?

Reply:
Yes, sounds good to me. I'll update it in the next commit.